cloud/hack/update-codegen.sh

```diff
@@ -20,7 +20,14 @@ set -o pipefail
 
 SCRIPT_ROOT=$(unset CDPATH && cd $(dirname "${BASH_SOURCE[0]}")/../.. && pwd)
 
-${SCRIPT_ROOT}/vendor/k8s.io/code-generator/generate-groups.sh "deepcopy" \
-github.com/kubeedge/kubeedge/cloud/pkg/client github.com/kubeedge/kubeedge/cloud/pkg/devicecontroller/apis \
+# generate the code with:
+# --output-base    because this script should also be able to run inside the vendor dir of
+#                  k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir
+#                  instead of the $GOPATH directly. For normal projects this can be dropped.
+${SCRIPT_ROOT}/vendor/k8s.io/code-generator/generate-groups.sh "deepcopy,client,informer,lister" \
```

Comment:
We are not using the client,informer,lister currently in our controllers, so these were not added to the code generation script. Also, since now we have upgraded to K8S 1.15, can we explore using kubebuilder for code generation. It also generates / updates the crd definition yamls which is currently not possible with the code-generator

Reply:
The client/informer/lister is useful for developers who develop applications that use kubeedge apis. Ideally we should offer them in an other repo, but to have an easy start, we can check them into the main repo first.